dagshub/upload/wrapper.py

```diff
@@ -26,19 +27,28 @@
 s.headers.update(config.requests_headers)
 
 
-def create_dataset(repo_name, local_path, glob_exclude="", org_name="", private=False):
+def create_dataset(repo_name,
+                   local_path,
+                   glob_exclude="",
+                   org_name="",
+                   private=False,
+                   repo=None,
+                   data_dir=DEFAULT_DATA_DIR_NAME):
     """
-    Create a new repository on DagsHub and upload an entire dataset to it
-
+    Create a new repository on DagsHub (or get an existing repo) and upload an entire dataset to it
     :param repo_name (str): Name of the repository to be created
     :param local_path (str): local path where the dataset to upload is located
     :param glob_exclude (str): regex to exclude certain files from the upload process
     :param org_name (str): Organization name to be the repository owner
     :param private (bool): Flag to indicate the repository is going to be private
+    :param repo_name (Reop): a previously created repository
+    :param data_dir (str): name of data directory that will be created inside repo
     :return : Repo object of the repository created
     """
-    repo = create_repo(repo_name, org_name=org_name, private=private)
-    dir = repo.directory(repo_name)
+    if not repo:
+        repo = create_repo(repo_name, org_name=org_name, private=private)
+
+    dir = repo.directory(data_dir)
     dir.add_dir(local_path, glob_exclude)
     return repo
```

Comment:
This changes the default behavior of `create_dataset` when no repo is passed. Now it expects the data to always be in the `data_dir`, instead of the repo name.

Maybe spin your behavior into an `add_dataset` function on the Repo object, because the logic here is already getting a bit confusing